### PR TITLE
Cocina-izes VersionService. Stores version data in db.

### LINKED ISSUE #3170: Rewrite VersionService and ObjectsController.accession to use Cocina objects

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3170

## Why was this change made?
Version metadata is being moved from the version datastream to the DB.

## How was this change tested?
Unit

### CODE REVIEW

File: app/controllers/versions_controller.rb [hunk 1]
```diff
@@ -1,45 +1,50 @@
 # frozen_string_literal: true
 
 class VersionsController < ApplicationController
-  before_action :load_item
+  before_action :load_cocina_object
 
   def index
-    return render json: {} unless @item.respond_to?(:versionMetadata)
+    # This can be removed after migration.
```
Comment (jcoyne) [inline app/controllers/versions_controller.rb:7]: Do we need to keep a list of what has and what hasn't migrated?

Comment (justinlittman) [reply app/controllers/versions_controller.rb:7]: My thought was to create an Uber Migration job that would migrate versions, events, and persist to cocina and then invoke that for every object. So I don't think there is a need for a separate tracking mechanism for versions.

**Comment by jcoyne** [comment]
Holding until when?

**Comment by justinlittman** [comment]
Testing completed, so ready for review / merge.

File: spec/requests/authorization_spec.rb [hunk 2]
```diff
@@ -3,10 +3,10 @@
 require 'rails_helper'
 
 RSpec.describe 'Authorization' do
-  let(:object) { instance_double(Dor::Item, current_version: '5') }
+  let(:cocina_object) { instance_double(Cocina::Models::DRO, version: '5') }
```
Comment (jcoyne) [inline spec/requests/authorization_spec.rb:6]: Version should be an integer. https://github.com/sul-dlss/cocina-models/blob/main/openapi.yml#L197-L198

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Review by jcoyne — CHANGES_REQUESTED** [review]
Shouldn't this line have been updated too? https://github.com/sul-dlss/dor-services-app/blob/main/app/services/embargo_release_service.rb#L43

**Comment by justinlittman** [comment]
@jcoyne Yes, changes are needed to EmbargoReleaseService. In fact, I made them. And then I forgot to hit save in the editor. That showed up here as well: https://app.honeybadger.io/projects/50568/faults/83969890

**Review by jcoyne — APPROVED** [review]

**Pull request merged**